types/nwmatcher/nwmatcher-tests.ts

```diff
@@ -0,0 +1,55 @@
+import nwmatcher = require('nwmatcher');
+
+/**
+ * Test matcher create function from module.
+ */
+
+const global: { document: Document } = { document: {} as any };
+let matcher = nwmatcher(global);
+
+/**
+ * Test matcher (select/first/...) functions.
+ */
+
+const element: Element = {} as any;
+
+let resultElementOrNull: Element | null;
+let resultStringOrUndefined: string | undefined;
+let resultElementArray: Element[];
+let resultBoolean: boolean;
+
+resultElementOrNull = matcher.first('foo');
+resultElementOrNull = matcher.first('foo', element);
+resultElementOrNull = matcher.first('foo');
+resultElementOrNull = matcher.first('foo', element);
+
+resultBoolean = matcher.match(element, 'foo');
+resultBoolean = matcher.match(element, 'foo', element);
+
+resultElementArray = matcher.select('foo');
+resultElementArray = matcher.select('foo', element);
+resultElementArray = matcher.select('foo', element, (element: Element) => undefined);
+
+resultElementOrNull = matcher.byId('foo');
+resultElementOrNull = matcher.byId('foo', element);
+resultElementOrNull = matcher.byId('foo');
+resultElementOrNull = matcher.byId('foo', element);
+
+resultElementArray = matcher.byTag('foo');
+resultElementArray = matcher.byTag('foo', element);
+
+resultElementArray = matcher.byClass('foo');
+resultElementArray = matcher.byClass('foo', element);
+
+resultElementArray = matcher.byName('foo');
+resultElementArray = matcher.byName('foo', element);
+
+resultStringOrUndefined = matcher.getAttribute(element, 'foo');
+
+resultBoolean = matcher.hasAttribute(element, 'foo');
+
+/**
+ * Test global namespace.
+ */
+
+matcher = NW.Dom;
```

Comment:
You want global test in separate file YourLibraryName-global.test.ts as per https://github.com/DefinitelyTyped/DefinitelyTyped#how-do-i-write-definitions-for-packages-that-can-be-used-globally-and-as-a-module